lxd/storage/drivers/driver_ceph_volumes.go

```diff
@@ -175,6 +175,33 @@ func (d *ceph) CreateVolume(vol Volume, filler *VolumeFiller, op *operations.Ope
 		if err != nil {
 			return err
 		}
+
+		if vol.contentType == ContentTypeBlock {
+			// Re-create the readonly snapshot, post-filling.
```

Comment:
@monstermunchkin why does the snapshot need to be removed and re-created, as the filling happens before the initial snapshot on line 167? What problem did you encounter that lead to this change? Thanks

Reply:
@tomponline The snapshot is missing the `metadata.yaml` and the `template` directory. The initial snapshot is created before the actual image is filled which makes it empty and unusable. The `zfs` driver does the same thing.

Reply:
@monstermunchkin right I see, you mean that the snapshot of the config volume that occurs as part of line 119, which is before the filler function runs on line 139, so once the block volume's filler function has run (which can also unpack files into the config volume) we then need to remove the config volume's snapshot and re-create it as part of creating the block volume.

It would be good to draw attention to this in the comment on line 180 (and equivalent comment in ZFS driver too), that we are talking about the FS config volume and not the volume the function is currently being worked on.

Something like "// Re-create the FS config volume's readonly snapshot now that the filler function has run and unpacked into both config and block volumes".

Thanks